.github/workflows/Test_for_levelup.yml

```diff
@@ -0,0 +1,25 @@
+name: Static code analysis including build test and memory check
+
+on:
+  push
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install cppcheck and valgrind
+        run: sudo apt install -y cppcheck && sudo apt install -y valgrind 
+      - name: Run code analysis with cppcheck
+        run: cppcheck *.cpp --enable=all --output-file=cppcheck_output.txt && chmod +x check_for_warning.sh && ./check_for_warning.sh && chmod +x check_for_error.sh && ./check_for_error.sh
+      - name: Upload Cppcheck result 
+        uses: actions/upload-artifact@v2  
+        with:
+          name: cppcheck_output
+          path: cppcheck_output.txt
+      - name: Build
+        run: g++ -Wall -Werror -Wextra -std=c++17 *.cpp -o a.out
+      - name: Run Valgrind check
+        run: valgrind --error-exitcode=1 ./a.out units/Luke.json units/Vader.json
```

Comment:
A valgrindnak még meg kell adni a `--leak-check=full` opciót is, különben nem minden hibára fog a megadott error-exitcode-dal visszatérni.

Reply:
Köszönöm, javítottam!